controllers/ptpoperatorconfig_controller.go

```diff
@@ -186,6 +191,62 @@ func (r *PtpOperatorConfigReconciler) setDaemonNodeSelector(
 	return obj, nil
 }
 
+// syncPvc update PersistentVolumeClaim
+func (r *PtpOperatorConfigReconciler) syncPvc(ctx context.Context, obj *uns.Unstructured) (*uns.Unstructured, error) {
+	var err error
+	scheme := kscheme.Scheme
+	pvc := &corev1.PersistentVolumeClaim{}
+	err = scheme.Convert(obj, pvc, nil)
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert obj to corev1.PersistentVolumeClaim: %v", err)
+	}
+
+	// update the VolumeName of PVC when the PVC is bound to a PV
+	if pvcDeployed := r.getPvc(obj.GetName(), names.Namespace); pvcDeployed != nil {
+		if pvcDeployed.Spec.VolumeName != pvc.Spec.VolumeName && pvc.Spec.VolumeName == "" {
+			log.Printf("pvc %s is Bound, updating VolumeName to %s", obj.GetName(), pvcDeployed.Spec.VolumeName)
+			pvc.Spec.VolumeName = pvcDeployed.Spec.VolumeName
+		}
+	}
+
+	err = scheme.Convert(pvc, obj, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert corev1.PersistentVolumeClaim to obj: %v", err)
+	}
+
+	return obj, nil
+}
+
+// cleanupPvc clean up obsolete PVCs not mounted to current lixnuxptp pod
+func (r *PtpOperatorConfigReconciler) cleanupPvc(ctx context.Context, storageType string) error {
```

Comment:
for multi node support (after you fixed it ) , wildchar the  node_name section of it